### ISSUE TITLE
aws eks private link support

### ISSUE THREAD

**Issue by stijndehaes** [terraform-providers/terraform-provider-aws#8022]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Private link support to eks got added very recently see following [issue](https://github.com/aws/containers-roadmap/issues/22).

I would like to configure this from terraform.

### New or Affected Resource(s)

* resource_aws_eks_cluster

### Potential Terraform Configuration

I would add this to the vpc config of the aws_eks_resource something like:

```hcl
resource "aws_eks_cluster" "this" {
  name     = "${var.cluster_name}"
  role_arn = "${aws_iam_role.cluster.arn}"
  version  = "${var.cluster_version}"

  vpc_config {
    security_group_ids = ["${local.cluster_security_group_id}"]
    subnet_ids         = ["${var.subnets}"]
    private_access = true
    public_access = true
  }

  timeouts {
    create = "${var.cluster_create_timeout}"
    delete = "${var.cluster_delete_timeout}"
  }

}
```

### References

* https://github.com/aws/containers-roadmap/issues/22

**Comment by bflad** [comment]
Work in progress pull request can be found here: https://github.com/terraform-providers/terraform-provider-aws/pull/8024

**Comment by ewbankkit** [comment]
AWS announcement:
* https://aws.amazon.com/about-aws/whats-new/2019/03/amazon-eks-introduces-kubernetes-api-server-endpoint-access-cont/

**Comment by bflad** [comment]
Support for managing the EKS Cluster endpoint access has been merged and will release with version 2.3.0 of the Terraform AWS Provider, likely in the next day or two.

**Comment by paulkendobnc** [comment]
Wonderful! Just what i need right now.

**Comment by stephenmuss** [comment]
That was fast. Looking forward to this. Thanks!

**Comment by bflad** [comment]
This has been released in [version 2.3.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#230-march-21-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
